### PR TITLE
JID3314-58 Subtask 2/2: Displaying Statistics at the Top of Each Report

### PR DESCRIPTION
**Pull request by waffy1901**

### CODE REVIEW

**Review by jonnamorrison — APPROVED** [review]
great job!

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 1]
```diff
@@ -2,11 +2,32 @@ import { useState, useEffect } from "react"
...
 export default function Report({ reportID }) {
   const [results, setResults] = useState(null)
+  const [statistics, setStatistics] = useState(null)
+  const [showDiseaseStats, setShowDiseaseStats] = useState(false)
 
   useEffect(() => {
     // Here you will fetch the report data from the API based on the reportID
+    const fetchReportStatistics = async () => {
+		try {
+			const response = await fetch(`http://localhost:8000/report_statistics/${reportID}`)
```
Comment (solomonmcharo28) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:12]: may not want to hardcode the endpoint, could use a global variable to identify the beginning of the endpoint as its gonna change when everything goes live

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 2]
```diff
@@ -2,11 +2,32 @@ import { useState, useEffect } from "react"
...
+		fetchReportStatistics()
+	}
   }, [reportID])
 
+  const toggleDiseaseStats = () => {
+	setShowDiseaseStats(!showDiseaseStats)
+  }
+
   const handleDownload = (e) => {
     const csvData =
```
Comment (solomonmcharo28) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:32]: this variable is never used again

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 3]
```diff
@@ -37,6 +58,69 @@ export default function Report({ reportID }) {
...
+			<>
+			<button
+				type="button"
+				className="bg-blue-400 text-white px-5 py-2 rounded-md hover:bg-blue-600 mb-4"
+				onClick={toggleDiseaseStats}
+			>
+				{showDiseaseSpecificStats ? "Hide" : "Show"} Disease Specific Stats
+			</button>
+
+			{!showDiseaseSpecificStats ? (
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:73]: showDiseaseSpecificStats is not a defined variable. I think you meant to use showDiseaseStats

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 4]
```diff
@@ -37,6 +58,69 @@ export default function Report({ reportID }) {
...
+					<th>Total Cases</th>
+					<th>Total Duplicates</th>
+					<th>Total Missing From CDC</th>
+					<th>Total Missing From States</th>
+					<th>Total Wrong Attributes</th>
+					</tr>
+				</thead>
+				<tbody>
+					<tr>
+					<td>{statistics.TotalCases}</td>
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:86]: Make sure to actually calculate the totals for the entire report. There does not exist TotalCases, TotalDuplicates, etc.. attributes on the statistics data. You will need to calculate this whenever you get the data back from the server. You can either calculate the total statistics for the report either server side using SQL or on the frontend.

File: CDC-Data-Reconciliation-Frontend/src/components/Report.jsx [hunk 5]
```diff
@@ -37,6 +58,69 @@ export default function Report({ reportID }) {
...
+					<th>Event Code</th>
+					<th>Total Cases</th>
+					<th>Total Duplicates</th>
+					<th>Total Missing From CDC</th>
+					<th>Total Missing From States</th>
+					<th>Total Wrong Attributes</th>
+					</tr>
+				</thead>
+				<tbody>
+					{statistics.DiseaseStats.map((stat) => (
```
Comment (Carson21) [inline CDC-Data-Reconciliation-Frontend/src/components/Report.jsx:107]: There does not exist a DiseaseStats attribute on the statistics data you get back from the API. you should just use statistics.map

**Review by Carson21 — CHANGES_REQUESTED** [review]
Make sure to calculate the total statistics for the report.

**Review by Carson21 — APPROVED** [review]
Looks good now

**Pull request merged**